NMS-13971: Basic upgrade procedure (#4606)
OpenNMS/opennms: docs/modules/deployment/pages/upgrade/rhel/up-rhel.adoc

Your {page-component-title} repository is defined in a file in `/etc/yum.repos.d/`.

ifeval::["{page-component-title}" == "Horizon"]
It may be named `opennms-repo-stable-<OSversion>.repo but is not guaranteed to be.
endif::[]

ifeval::["{page-component-title}" == "Meridian"]
It may be named {page-component-title}.repo but is not guaranteed to be.
endif::[]

. Become root.

> . Use the text editor of your choice to edit your {page-component-title} repository config to change the `baseurl` of the repository from the old release family to the new release family.

Reviewer: This differs between Horizon and Meridian...  Horizon repo tracks `stable` which is always the current release AFAIK, meridian repo is by release-family year (2019, 2020, etc).  Meridian would have to change the `baseUrl` to move between major releases, Horizon never has to change the `baseUrl`.

> . Use the text editor of your choice to edit your {page-component-title} repository config to change the `baseurl` of the repository from the old release family to the new release family.

Reviewer 2: Probably don't need this step on Horizon, but may need this on Meridian.

. Enable auto updates:
+
[source, console]
----
yum -y install yum-utils

> yum-config-manager --enable opennms-repo-stable-*

Reviewer 2: added suggestion for this step

----

. Purge any cached yum data:
+
[source, console]
----
yum clean all
----

. Make a backup copy of your OpenNMS config:

+
[source, console]
----

> rsync -Ppav /opt/opennms/etc /tmp/etc.orig
> rsync -Ppav /opt/opennms/jetty-webapps/opennms/WEB-INF /tmp/opennms-web-inf

Reviewer 2: ```suggestion
rsync -Ppav $\{OPENNMS_HOME}/etc /tmp/etc.orig
rsync -Ppav$\{OPENNMS_HOME}/jetty-webapps/opennms/WEB-INF /tmp/opennms-web-inf
```

----

. Upgrade the {page-component-title} packages to the new version:
+
[source, console]
----
yum -y upgrade opennms
----

> 

Reviewer 2: ```suggestion

. Disable auto updates:
+
[source, console]
----
yum-config-manager --disable opennms-repo-stable-*
----

```

> . Install Java 11:

Reviewer 2: ```suggestion
. Upgrade Java 11 to the latest release:
```

+
[source, console]
----
yum -y install java-11-openjdk java-11-openjdk-devel
----

> . Execute runjava:

Reviewer 2: ```suggestion
. Execute `runjava` to update which JVM {page-component-title} will use:
```

+
[source, console]
----

> $OPENNMS_HOME/bin/runjava -s

Reviewer 2: ```suggestion
$\{OPENNMS_HOME}/bin/runjava -s
```

----

. Check for configuration file changes, and update accordingly, using the files you backed up in xref:deployment:upgrade/diff.adoc#run_diff[Idenfity Changed Configuration Files].
+

> NOTE: If you upgrade in place, OpenNMS renames any shipped config that conflicts with an existing user-modified config to `.rpmnew`.

Reviewer 2: ```suggestion
NOTE: If you upgrade in place, OpenNMS renames any shipped config that conflicts with an existing user-modified config to `.rpmnew` or `.rpmsave`.
```

Inspect these files manually and reconcile the differences.
Use `diff -Bbw` and `diff -y`.

> 

Reviewer 2: ```suggestion
If any `.rpmnew` or `.rpmsave` file exist within the configuration directory, services will not start.

```

> . Run the opennms installer:

Reviewer 2: ```suggestion
. Run the {page-component-title} installer:
```

+
[source, console]
----

> $OPENNMS_HOME/bin/install -dis

Reviewer 2: ```suggestion
$\{OPENNMS_HOME}/bin/install -dis
```

----
+

The upgrade may take some time.
Eventually the following message appears: `Upgrade completed successfully!`

. Clear the Karaf cache:
+
[source, console]
----

> yes | $OPENNMS_HOME/bin/fix-karaf-setup.sh

Reviewer 2: ```suggestion
yes | $\{OPENNMS_HOME}/bin/fix-karaf-setup.sh
```

----

. Start OpenNMS {page-component-title}:
+
[source, console]
----
systemctl start opennms.service
----
+

> NOTE: `tail -F /var/log/opennms/manager.log` can illustrate the current point in the startup process {page-component-title} is.

Reviewer 2: ```suggestion
NOTE: `tail -F $\{OPENNMS_HOME}/logs/manager.log` can illustrate the current point in the startup process {page-component-title} is.
```

systemctl start opennms.service
----
+

> NOTE: `tail -F $\{OPENNMS_HOME}/logs/manager.log` can illustrate the current point in the startup process {page-component-title} is.

Reviewer 2: ```suggestion
NOTE: `tail -F $\{OPENNMS_HOME}/logs/manager.log` can illustrate the current point in the startup process {page-component-title} is.
```
